packages/shared-ux/error_boundary/src/ui/section_error_boundary.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import React from 'react';
+
+import type { KibanaErrorBoundaryServices } from '../../types';
+import { useErrorBoundary } from '../services/error_boundary_services';
+import { SectionErrorPrompt } from './message_components';
+
+interface ErrorBoundaryState {
+  error: null | Error;
+  errorInfo: null | Partial<React.ErrorInfo>;
+  componentName: null | string;
+}
+
+interface SectionErrorBoundaryProps {
+  sectionName: string;
+}
+
+interface ServiceContext {
+  services: KibanaErrorBoundaryServices;
+}
+
+class SectionErrorBoundaryInternal extends React.Component<
+  React.PropsWithChildren<SectionErrorBoundaryProps> & ServiceContext,
+  ErrorBoundaryState
+> {
+  constructor(props: SectionErrorBoundaryProps & ServiceContext) {
+    super(props);
+    this.state = {
+      error: null,
+      errorInfo: null,
+      componentName: null,
+    };
+  }
+
+  componentDidCatch(error: Error, errorInfo: React.ErrorInfo) {
+    console.error('Error caught by Kibana React Error Boundary'); // eslint-disable-line no-console
+    console.error(error); // eslint-disable-line no-console
+
+    const { name } = this.props.services.errorService.registerError(error, errorInfo);
+    this.setState({ error, errorInfo, componentName: name });
+  }
+
+  render() {
+    if (!this.state.error) {
+      return this.props.children;
+    }
+
+    const { error, errorInfo, componentName } = this.state;
+
+    return (
+      <SectionErrorPrompt
```

Comment:
Isn't it possible the caught error could be a non-fatal error, such as an async chunk that failed to load because of a temporary network issue? Note how when errors are caught in the `ErrorBoundaryInternal` class in `packages/shared-ux/error_boundary/src/ui/error_boundary.tsx` file, we evaluate the type of error and show a different type of message for "non-fatal" errors.